### PR TITLE
Deploy server as an action and configure dependencies via newer pipenv

### PR DESCRIPTION
**Pull request by avan1235**
It's configured to refresh the data on server every time the push changes the django-server folder at any branch (so clean the server repo on heroku and push there all the files from server folder to setup server based on Pipfile)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ __pycache__/
 django-server/server/db.sqlite3
 .env
 django-server/serverenv/*
+/django-server/server/static/
```
Comment (kantoniak) [inline .gitignore]: Extra `/` (start of the line)?

File: django-server/server/Procfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+release: python manage.py makemigrations; python manage.py migrate --no-input
+web: gunicorn server.wsgi
```
Comment (kantoniak) [inline django-server/server/Procfile]: Missing newline

File: django-server/server/api/models.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+
+
+class Cafeteria(models.Model):
+    name = models.CharField(max_length=128)
+    description = models.CharField(max_length=256)
+    sub_description = models.CharField(max_length=512)
+    latitude = models.FloatField()
+    longitude = models.FloatField()
+    capacity = models.IntegerField()
```
Comment (kantoniak) [inline django-server/server/api/models.py]: Missing validator for minimum value as in API.

File: django-server/server/api/views.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from rest_framework import viewsets
+
+from .serializers import CafeteriaSerializer
+
+from .models import Cafeteria
+
+
+class CafeteriaViewSet(viewsets.ModelViewSet):
+    queryset = Cafeteria.objects.all().order_by('id')
+    serializer_class = CafeteriaSerializer
```
Comment (kantoniak) [inline django-server/server/api/views.py]: Capacity is a private property and should not appear in get requests for all. Can we have different serializers for public and private properties? It would be best to show all properties to roles "cafeteria owner" and "admin".

File: django-server/server/server/.env.sample [hunk 5]
```diff
@@ -1,5 +0,0 @@
-USE_POSTGRES=False
```
Comment (kantoniak) [inline django-server/server/server/.env.sample]: Why is this file deleted?

File: django-server/server/server/settings.py [hunk 6]
```diff
@@ -76,12 +78,23 @@
...
+}
+
+
 # Database
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
 
-
 def select_database():
-    if env.bool('USE_POSTGRES', default=False):
+    if env.bool('DB_DEPLOY', default=False):
```
Comment (kantoniak) [inline django-server/server/server/settings.py]: We can still deploy to PostgreSQL locally, right?

File: django-server/server/server/urls.py [hunk 7]
```diff
@@ -16,7 +16,9 @@
 from django.contrib import admin
 from django.urls import include, path
 
+
 urlpatterns = [
     path('', include('home.urls')),
     path('admin/', admin.site.urls),
+    path('api/', include('api.urls')),
```
Comment (kantoniak) [inline django-server/server/server/urls.py]: Can we use `api/beta/` in here? It's a good practice to have API version in URL.

File: django-server/server-env-setup.sh [hunk 8]
```diff
@@ -1,12 +1,9 @@
...
-python3 -m venv serverenv
-source serverenv/bin/activate
-pip install django
-pip install django-environ
-pip install djangorestframework
+# This script assumes you have python3, pip and pipenv installed
+# you can install pipenv using pip3 install pipenv
+echo "Starting installation of virtual environment"
+cd server
+pipenv install
```
Comment (kantoniak) [inline django-server/server-env-setup.sh]: Is it possible to configure virtualenv name? To me it seems env will use some default.

File: django-server/server/server/settings.py (hunk 6 above)
Comment (avan1235) [reply django-server/server/server/settings.py]: yeah, just define env DB_DEPLOY=True and the env with database auth

File: django-server/server-env-setup.sh (hunk 8 above)
Comment (avan1235) [reply django-server/server-env-setup.sh]: why do you need it? let's see https://github.com/pypa/pipenv/issues/1226. to use env later just type pipenv shell

File: .gitignore (hunk 1 above)
Comment (avan1235) [reply .gitignore]: fixed

File: django-server/server/Procfile (hunk 2 above)
Comment (avan1235) [reply django-server/server/Procfile]: fixed

File: django-server/server/api/models.py (hunk 3 above)
Comment (avan1235) [reply django-server/server/api/models.py]: to be discussed: cannot set constraint in django that will generate sql with needed checks on database, only write a validator that can check this value on server (fixed in this way). Have to check if this works with API generators

File: django-server/server/server/.env.sample (hunk 5 above)
Comment (avan1235) [reply django-server/server/server/.env.sample]: as for me it was a sample how to configure  the env on server and is no longer needed

File: django-server/server/server/urls.py (hunk 7 above)
Comment (avan1235) [reply django-server/server/server/urls.py]: good idea, changed

File: django-server/server/api/views.py (hunk 4 above)
Comment (avan1235) [reply django-server/server/api/views.py]: marked to be discussed in the near  future, when it will be obvious how to configure permissions

Comment (avan1235) [reply django-server/server/api/views.py]: marked as TODO in code in serializers of the apu

File: django-server/server/server/.env.sample (hunk 5 above)
Comment (Rhantolq) [reply django-server/server/server/.env.sample]: .env is no longer needed? If it is needed, then why delete .sample?

Comment (avan1235) [reply django-server/server/server/.env.sample]: env is defined by the environmnent variables on heroku in system. sample back again to repo

File: .github/workflows/server_deploy.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+name: Webserver deploy
+
```
Comment (kantoniak) [inline .github/workflows/server_deploy.yaml]: Maybe from `master` only? People would steal Heroku from each other on multiple branches.

File: django-server/server-env-setup.sh (hunk 8 above)
Comment (kantoniak) [reply django-server/server-env-setup.sh]: We should minimize changes. This one seems fine.

File: django-server/server/api/views.py (hunk 4 above)
Comment (kantoniak) [reply django-server/server/api/views.py]: Opened #55 so we don't miss that.

File: .github/workflows/server_deploy.yaml (hunk 9 above)
Comment (avan1235) [reply .github/workflows/server_deploy.yaml]: but it is deployed there only if changes to django server are made. I think that it would be easier in bta version when we would like to test the api

**Comment by avan1235** [comment]
can introduce testing deploy server as the separate action, then will change this action to be launched only on master branch
